presto-test-jdbc-compatibility-old-server/src/test/java/io/prestosql/TestJdbcResultSetCompatibility.java

```diff
@@ -0,0 +1,125 @@
+import io.prestosql.jdbc.PrestoDriver;
+import org.testcontainers.containers.PrestoContainer;
+import org.testng.SkipException;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+@Test(singleThreaded = true)
+public class TestJdbcResultSetCompatibility
+        extends BaseTestJdbcResultSet
+{
+    private static final int NUMBER_OF_TESTED_VERSIONS = 5;
+    private static final int TESTED_VERSIONS_GRANULARITY = 3;
+
+    private final Optional<String> testedPrestoVersion;
+    private PrestoContainer<?> prestoContainer;
+
+    @Factory(dataProvider = "testedPrestoVersions")
+    public TestJdbcResultSetCompatibility(Optional<String> testedPrestoVersion)
+    {
+        this.testedPrestoVersion = requireNonNull(testedPrestoVersion, "testedPrestoVersion is null");
+    }
+
+    @DataProvider
+    public static Object[][] testedPrestoVersions()
+    {
+        try {
+            String currentVersionString = PrestoDriver.class.getPackage().getImplementationVersion();
+            if (currentVersionString == null) {
+                return new Object[][] {
+                        {Optional.empty()}
+                };
+            }
+
+            int currentVersion = Integer.parseInt(Splitter.on("-").omitEmptyStrings().trimResults().splitToList(currentVersionString).get(0));
+            ImmutableList.Builder<String> testedPrestoVersions = ImmutableList.builder();
+            int testedVersion = currentVersion - 1;
+            for (int i = 0; i < NUMBER_OF_TESTED_VERSIONS; i++) {
+                testedPrestoVersions.add(String.valueOf(testedVersion));
+                testedVersion -= TESTED_VERSIONS_GRANULARITY;
+            }
+
+            return testedPrestoVersions.build().stream()
+                    .map(version -> new Object[] {Optional.of(version)})
+                    .toArray(Object[][]::new);
+        }
+        catch (Throwable e) {
+            System.err.println("Could not determine Presto versions to test; " + e.getMessage() + "\n" + getStackTrace(e));
+            return new Object[][] {
+                    {Optional.empty()}
+            };
+        }
+    }
+
+    private static String getStackTrace(Throwable e)
+    {
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter printWriter = new PrintWriter(stringWriter);
+        e.printStackTrace(printWriter);
+        return stringWriter.toString();
+    }
+
+    @BeforeClass
+    public void setupPrestoContainer()
+    {
+        if (testedPrestoVersion.isEmpty()) {
+            throw new AssertionError("Could not determine current Presto version");
+        }
+        Logging.initialize();
```

Comment:
Why?

Reply:
I do not recall adding it explicitly but we do data in a lot of tests. I can drop. We probably do not need logging in this VM as presto server is run from docker.